### ISSUE TITLE
The pretrained model is only trained on SAM 1B dataset. So it cannot output the segmeantion mask with class lablel right?

### ISSUE THREAD

**Issue by guanguanboy** [UX-Decoder/Semantic-SAM#63]

**Comment by guanguanboy** [comment]
Who can respond?

**Comment by XinyingZheng** [comment]
你问题解决了么，这个一般都是交互式的，是不是只能自己手动修改啊，比如SEEM工程，支持文本prompt，但看上去要改的东西挺多

**Comment by XinyingZheng** [comment]
网络不好，评论删不掉了。。看了代码挺好改的，没一会就已经改好了

**Comment by PadenU** [comment]
> 网络不好，评论删不掉了。。看了代码挺好改的，没一会就已经改好了

想请教一下老哥怎么改的呀

**Comment by XinyingZheng** [comment]
我用的SEEM工程，你看一下交互式用的那个gradio那个库，只需要前面的部分，保存好推理的结果，把那个gradio交互式的部分去掉就可以了，还有，我是老妹，哈哈




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "UX-Decoder/Semantic-SAM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月16日(星期四) 凌晨1:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [UX-Decoder/Semantic-SAM] The pretrained model is only trained on SAM 1B dataset. So it cannot output the segmeantion mask with class lablel right? (Issue #63)





  
网络不好，评论删不掉了。。看了代码挺好改的，没一会就已经改好了
  
想请教一下老哥怎么改的呀

**Comment by vietpho** [comment]
@XinyingZheng 
Hello,
Sorry for interrupting—I can't speak Chinese and had to use a translator to read your conversation. 
Could you explain in more detail about the changes you made, or would you be able to share the code?
I'm quite new to the field of deep learning, so apologies for any inconvenience!

**Comment by PadenU** [comment]
> 我用的SEEM工程，你看一下交互式的梯度那个库，只需要前面的部分，保留好的推理结果，把梯度交互式的部分去掉就可以了，还有，我是老妹，哈哈
> [……](#)
> ------------------ 原始邮件 -------------------- 发件人: "UX-Decoder /语义-SAM"***@***.***>; 发送时间：2023年11月16日（周四）凌晨1:14 ***@***.***>；**@***.******@***.***>; 主题: 回复：[UX-Decoder/Semantic-SAM] 预训练模型仅在 SAM 1B 数据集上进行训练。所以它不能输出带有类标签的分段掩码，对吗？（[第63](https://github.com/UX-Decoder/Semantic-SAM/issues/63)期）网络不好，评论删不掉了。看了代码一大改的，没一会就已经改好了想请教一下老哥怎么改的呀 — 直接回复这封邮件，查看一下在 GitHub 上，或取消订阅。您收到此消息是因为您发表了评论。消息 ID：***@***.***>

感谢回复，才看见，我也弄好了hhhhh，感恩老妹！

**Comment by Chase-Arthur** [comment]
> 我用的SEEM工程，你看一下交互式用的那个gradio那个库，只需要前面的部分，保存好推理的结果，把那个gradio交互式的部分去掉就可以了，还有，我是老妹，哈哈
> […](#)

@XinyingZheng 哈喽您好！请问是否能问一下您的修改细节，您说的"只需要前面的部分，保存好推理的结果，把那个gradio交互式的部分去掉"具体指的是什么？如果可以的话，能否加一下QQ交流一下？（QQ：1084734352）万分感谢！

**Comment by lijian181** [comment]
> [@XinyingZheng](https:github.com/XinyingZheng) Hello,  你好 Sorry for interrupting—I can't speak Chinese and had to use a translator to read your conversation.很抱歉打扰了——我不会说中文，不得不让翻译来阅读你的对话。 Could you explain in more detail about the changes you made, or would you be able to share the code?您能否更详细地解释一下您所做的更改，或者您能否共享代码？ I'm quite new to the field of deep learning, so apologies for any inconvenience!我对深度学习领域还很陌生，所以对于给您带来的不便，我们深表歉意！



hey，guys。 I've resolved the issue by primarily modifying two files: task/automatic_mask_generator.py and tsask/interactive_idino_m2m_auto.py.
Modifications in tsask/interactive_idino_m2m_auto.py:
Commented out the code following fig... in the interactive_infer_image function and directly return outputs.
Modifications in task/automatic_mask_generator.py:
Adjusted the MaskData structure by saving a copy of masks before the del masks operation.
This approach ensures the necessary data is preserved while streamlining the inference process.    I've marked the newly added parts as modified. The run_semanticsam_logits.py can be used to simply test whether the modification is successful.

[automatic_mask_generator.txt](https://github.com/user-attachments/files/20993286/automatic_mask_generator.txt)
[interactive_idino_m2m_auto.txt](https://github.com/user-attachments/files/20993285/interactive_idino_m2m_auto.txt)
[run_semanticsam_logits.txt](https://github.com/user-attachments/files/20993287/run_semanticsam_logits.txt)
